disk/datadog_checks/disk/disk.py

```diff
@@ -147,47 +127,88 @@ def collect_metrics_psutil(self):
 
         self.collect_latency_metrics()
 
-    def _exclude_disk_psutil(self, part):
+    def exclude_disk(self, part):
         # skip cd-rom drives with no disk in it; they may raise
         # ENOENT, pop-up a Windows GUI error for a non-ready
         # partition or just hang;
         # and all the other excluded disks
         skip_win = Platform.is_win32() and ('cdrom' in part.opts or part.fstype == '')
         return skip_win or self._exclude_disk(part.device, part.fstype, part.mountpoint)
 
-    def _exclude_disk(self, name, filesystem, mountpoint):
+    def _exclude_disk(self, device, file_system, mount_point):
         """
         Return True for disks we don't want or that match regex in the config file
         """
-        self.log.debug('_exclude_disk: {}, {}, {}'.format(name, filesystem, mountpoint))
+        self.log.debug('_exclude_disk: {}, {}, {}'.format(device, file_system, mount_point))
+
+        if not device or device == 'none':
+            device = None
+
+            # Allow no device if `all_partitions` is true so we can evaluate mount points
+            if not self._all_partitions:
+                return True
 
         # Hack for NFS secure mounts
         # Secure mounts might look like this: '/mypath (deleted)', we should
-        # ignore all the bits not part of the mountpoint name. Take also into
-        # account a space might be in the mountpoint.
-        mountpoint = mountpoint.rsplit(' ', 1)[0]
+        # ignore all the bits not part of the mount point name. Take also into
+        # account a space might be in the mount point.
+        mount_point = mount_point.rsplit(' ', 1)[0]
 
-        name_empty = not name or name == 'none'
+        return (
+            self._disk_blacklisted(device, file_system, mount_point)
+            or not self._disk_whitelisted(device, file_system, mount_point)
+        )
 
-        # allow empty names if `all_partitions` is `true` so we can evaluate mountpoints
-        if name_empty and not self._all_partitions:
-            return True
-        # device is listed in `excluded_disks`
-        elif not name_empty and name in self._excluded_disks:
-            return True
-        # device name matches `excluded_disk_re`
-        elif not name_empty and self._excluded_disk_re.match(name):
+    def _disk_whitelisted(self, device, file_system, mount_point):
+        return (
+            self._whitelist_file_system(file_system)
+            and self._whitelist_device(device)
+            and self._whitelist_mount_point(mount_point)
+        )
+
+    def _disk_blacklisted(self, device, file_system, mount_point):
+        return (
+            self._blacklist_file_system(file_system)
+            or self._blacklist_device(device)
+            or self._blacklist_mount_point(mount_point)
+        )
+
+    def _whitelist_file_system(self, file_system):
```

Comment:
Same comment as previously, let's adjust the naming of all of these methods to make it obvious that they return booleans. Forgot to mention in my last review